azure/services/aso/aso.go

```diff
@@ -175,6 +176,8 @@ func (r *reconciler[T]) CreateOrUpdateResource(ctx context.Context, spec azure.A
 	if labels == nil {
 		labels = make(map[string]string)
 	}
+	labels[clusterv1.ClusterNameLabel] = r.clusterName
```

Comment:
Is this label not already set somewhere else?

Reply:
No, the label previously did not exist on any of the ASO resources.